public/app/percona/backup/components/BackupInventory/BackupInventory.tsx

```diff
@@ -55,14 +57,36 @@ export const BackupInventory: FC = () => {
       {
         Header: actions,
         accessor: 'id',
-        Cell: ({ row }) => <BackupInventoryActions onBackup={onBackupClick} backup={row.original as Backup} />,
-        width: '80px',
+        Cell: ({ row }) => (
+          <BackupInventoryActions onRestore={onRestoreClick} onBackup={onBackupClick} backup={row.original as Backup} />
+        ),
+        width: '110px',
       },
     ],
     []
   );
   const styles = useStyles(getStyles);
 
+  const onRestoreClick = (backup: Backup) => {
+    setSelectedBackup(backup);
+    setRestoreModalVisible(true);
+  };
+
+  const handleClose = () => {
+    setSelectedBackup(null);
+    setRestoreModalVisible(false);
+    setBackupModalVisible(false);
+  };
+
+  const handleRestore = async (serviceId: string, locationId: string, artifactId: string) => {
+    try {
+      await BackupInventoryService.restore(serviceId, locationId, artifactId);
```

Comment:
shouldn't we use the cancel token from the other PR?

Reply:
Yep, indeed, but it's probably better to wait for that to be merged instead of cherry picking it here, what do you think?